### ISSUE TITLE
false positive nullability warning

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn#48354]
**Version Used**: 

master

**Steps to Reproduce**:

```csharp
#nullable enable

public class A
{
    public B? B { get; set; }
}

public class B
{
}

public static class C
{
    public static void M(A? a)
    {
        var x = a;
        var y = a?.B;
        if (y is null) return;
        // if x is null, then y is definitely null due to `?.` operator.
        // we can't reach this point if x is null.
        x.ToString();
    }
}
```

**Expected Behavior**:

No warnings

**Actual Behavior**:

    warning CS8602: Dereference of a possibly null reference.

**Comment by jcouv** [comment]
You're correct that we can't reach the `ToString` call when `x` is `null`, but the compiler can't know that without doing alias tracking (ie. knowing that `a` and `x` happen to be the same when the test `a?.B` takes place). Such analysis is out-of-scope of nullability analysis.

**Comment by tom-englert** [comment]
@Youssef1313 there is now a fix available: https://www.nuget.org/packages/Nullable.Extended.Analyzer/
